agent_app_sample_code/cookbook_utils/cookbook_config.py

```diff
@@ -0,0 +1,203 @@
+          return False
+      except (NotFound) as e:
+          print(f"\n`{self.uc_catalog_name}` does not exist, trying to create...")
+          try:
+              _ = w.catalogs.create(name=self.uc_catalog_name)
+              print(f"\nPASS: UC catalog `{self.uc_catalog_name}` was created")
+              return True
+          except PermissionDenied as e:
+              print(
+                  f"\nFAIL: `{self.uc_catalog_name}` does not exist, and no permissions to create.  Please provide an existing UC Catalog."
+              )
+              return False
+
+    def validate_or_create_uc_schema(self) -> bool:
+      w = WorkspaceClient()
+      # Create UC Schema if it does not exist, otherwise, raise an exception
+      try:
+          _ = w.schemas.get(full_name=f"{self.uc_catalog_name}.{self.uc_schema_name}")
+          print(f"\nPASS: UC schema `{self.uc_catalog_name}.{self.uc_schema_name}` exists")
+          return True
+      except PermissionDenied as e:
+          print(f"\n`{self.uc_catalog_name}.{self.uc_schema_name}` exists, but you do not have permissions to access.  Please pass a different `uc_schema_name` or get permissions to {self.uc_catalog_name}.{self.uc_schema_name}")
+          return False
+      except (NotFound) as e:
+          print(f"\n`{self.uc_catalog_name}.{self.uc_schema_name}` does not exist, trying to create...")
+          try:
+              _ = w.schemas.create(name=self.uc_schema_name, catalog_name=self.uc_catalog_name)
+              print(f"\nPASS: UC schema `{self.uc_catalog_name}.{self.uc_schema_name}` created")
+              return True
+          except PermissionDenied as e:
+              print(
+                  f"\nFAIL: `{self.uc_catalog_name}.{self.uc_schema_name}` does not exist, and no permissions to create.  Please provide an existing UC Schema within your UC Catalog {self.uc_catalog_name}."
+              )
+              return False
+            
+    def validate_or_create_mlflow_experiment(self) -> bool:
+      try:
+          mlflow.set_experiment(self.mlflow_experiment_name)
+          print(
+              f"\nPASS: Using MLflow experiment name `{self.mlflow_experiment_name}`."
+          )
+          return True
+      except Exception as e:
+          print(
+              f"\nFAIL: `{self.mlflow_experiment_name}` is not a valid directory for an MLflow experiment.  An experiment name must be an absolute path within the Databricks workspace, e.g. '/Users/<some-username>/my-experiment'.\n\nIf you tried to specify a directory, either remove the `mlflow_experiment_name` parameter to try the default value or manually specify a valid path for `mlflow_experiment_name` to `AgentCookbookConfig(...)`.\n\nIf you did not pass a value for `mlflow_experiment_name` and are seeing this message, pass a valid workspace directory for `mlflow_experiment_name` and try again."
```

Comment:
The error message here was also decent when I passed a bad experiment name:
```
RestException: INVALID_PARAMETER_VALUE: Got an invalid experiment name 'abcd/Users/sid.murching@databricks.com/agent_app_name_mlflow_experiment'. An experiment name must be an absolute path within the Databricks workspace, e.g. '/Users/<some-username>/my-experiment'.
```

Will push some suggestions to simplify this, since there are other potential causes of failure here e.g. `PermissionDenied` (so catching `Exception` + printing this new message may be inaccurate)

Reply:
Basically I'll just push a suggestion that follows the pattern mentioned above (it's helpful to print when we're trying to create a new resource etc, but we can just let the SDK/API tell us what happened during failures and push error message improvements in the backend if needed)